### PR TITLE
Feat/#60 edit user modal

### PR DESCRIPTION
**Pull request by ciaranschutte**
- add Edit User modal
- move form hook out
- move validations out and split into individual to be composable

### CODE REVIEW

File: client/components/pages/submissionSystem/modals/validations.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import * as yup from 'yup';
+import { PROGRAM_MEMBERSHIP_TYPES } from '../../../../global/constants/index';
```
[the remark this answers is not in the document]
Comment (joneubank) [reply client/components/pages/submissionSystem/modals/validations.js]: also, can we use an absolute path here? `global/constants`

File: client/components/pages/submissionSystem/modals/useFormHook.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      ...form,
+      data: data.map((field, i) => (i === index ? { ...field, [key]: val } : field)),
+    });
+  };
+
+  // set single error
+  const setError = ({ key, val, index = 0 }) => {
+    setForm({
+      ...form,
+      errors: errors.map((field, i) => (i === index ? { ...field, [key]: val } : field)),
```
[the remark this answers is not in the document]
Comment (ciaranschutte) [reply client/components/pages/submissionSystem/modals/useFormHook.js]: @joneubank @hlminh2000 multiple sections. poor naming using 'field'
```
[
  {firstName: 'please fill in firstname'}, // 1st section
  // list of errors to match the list of data
  {firstName: 'please fill in first name'} // 2nd section
]
```

[the remark this answers is not in the document]
Comment (joneubank) [reply client/components/pages/submissionSystem/modals/useFormHook.js]: I thought maybe after re-reading how the edit user form worked with sections...

How would this work with a form with only one section? always just reference errors with index=0?

[the remark this answers is not in the document]
Comment (ciaranschutte) [reply client/components/pages/submissionSystem/modals/useFormHook.js]: @joneubank yup ` const validateField = async ({ key, index = 0 }) => ....`

**Pull request merged**